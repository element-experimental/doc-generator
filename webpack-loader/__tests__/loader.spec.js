const docMdLoader = require('../doc-makrdown-loader')

const md = `
## Title 1
demo 1
:::demo demo content
\`\`\`html
<el-alert>x</el-alert>
<script>
export default {
  data() {
    return {
      x: 1
    }
  }
}
</script>
<style>
.el-alert {
  font-size: 20
}
</style>
\`\`\`
:::
`
test('docMdLoader', () => {
  const content = docMdLoader(md)
  expect(content).toMatchSnapshot()
})
