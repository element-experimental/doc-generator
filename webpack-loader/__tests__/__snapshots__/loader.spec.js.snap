// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docMdLoader 1`] = `
"
    <template>
      <div class=\\"element-doc\\">
        <h2 id=\\"title-1\\"><a class=\\"header-anchor\\" href=\\"#title-1\\" aria-hidden=\\"true\\">Â¶</a> Title 1</h2>
<p>demo 1</p>
<demo-block>
        <div><p>demo content</p>
</div>
        <element-demo0 />
        <demo-code>&lt;el-alert&gt;x&lt;/el-alert&gt;
&lt;script&gt;
export default {
  data() {
    return {
      x: 1
    }
  }
}
&lt;/script&gt;
&lt;style&gt;
.el-alert {
  font-size: 20
}
&lt;/style&gt;
</demo-code></demo-block>
      </div>
    </template>
    <script>
      export default {
        name: 'component-doc',
        components: {
          \\"element-demo0\\": (function demo() {
    
    var render = function() {
  var _vm = this
  var _h = _vm.$createElement
  var _c = _vm._self._c || _h
  return _c(\\"div\\", [_c(\\"el-alert\\", [_vm._v(\\"x\\")])], 1)
}
var staticRenderFns = []
render._withStripped = true

  
    const democomponentExport = {
  data() {
    return {
      x: 1
    }
  }
}
    return {
      ...democomponentExport,
      render,
      staticRenderFns
    }
  })(),
        }
      }
    </script>
    <style>.el-alert {
  font-size: 20
}</style>
  "
`;
